lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -979,19 +982,13 @@ def update_endpoint(self, relation: Relation = None) -> None:
         for relation in relations:
             self._set_endpoint_on_relation(relation)
 
-    def _set_endpoint_on_relation_change(self, event: RelationEvent) -> None:
-        self._set_endpoint_on_relation(event.relation)
-
     def _set_endpoint_on_relation(self, relation: Relation) -> None:
-        """Set the the remote_write endpoint on relations.
+        """Set the remote_write endpoint on relations.
 
         Args:
-            relation: Optional relation. If provided, only this relation will be
-                updated. Otherwise, all instances of the `prometheus_remote_write`
-                relation managed by this `PrometheusRemoteWriteProvider` will be
-                updated.
+            relation: The relation whose data to update.
         """
-        address = self._endpoint_address or self._get_relation_bind_address()
+        address = self._endpoint_address or socket.getfqdn()
```

Comment:
May I presume this works for cross model relations, given that self-monitoring may be implemented through cross model relations ?

Reply:
This is a Prometheus Lib, not a Grafana agent one.